File: src/csharp/2_CreationalPatterns/4_Singleton/Monostate.cs
using System.Collections.Generic;
using static System.Console;

namespace DotNetDesignPatternDemos.Creational.Singleton.Monostate
{
  public class ChiefExecutiveOfficer
  {
    private static string name;
    private static int age;

    public string Name
    {
      get => name;
      set => name = value;
    }

    public int Age
    {
      get => age;
      set => age = value;
    }

    public override string ToString()
    {
      return $"{nameof(Name)}: {Name}, {nameof(Age)}: {Age}";
    }
  }

  public class Demo
  {
    static void Main(string[] args)
    {
      var ceo = new ChiefExecutiveOfficer();
      ceo.Name = "Adam Smith";
      ceo.Age = 55;

      var ceo2 = new ChiefExecutiveOfficer();
      WriteLine(ceo2);
    }
  }
}

File: src/js/2_CreationalPatterns/4_Singleton/monostate.js
class ChiefExecutiveOfficer
{
  get name() { return ChiefExecutiveOfficer._name; }
  set name(value) { ChiefExecutiveOfficer._name = value; }

  get age() { return ChiefExecutiveOfficer._age; }
  set age(value) { ChiefExecutiveOfficer._age = value; }

  toString()
  {
    return `CEO's name is ${this.name} ` +
      `and he is ${this.age} years old.`;
  }
}
ChiefExecutiveOfficer._age = undefined;
ChiefExecutiveOfficer._name = undefined;

let ceo = new ChiefExecutiveOfficer();
ceo.name = 'Adam Smith';
ceo.age = 55;

let ceo2 = new ChiefExecutiveOfficer();
ceo2.name = 'John Gold';
ceo2.age = 66;

console.log(ceo.toString());
console.log(ceo2.toString());